modin/experimental/engines/omnisci_on_native/frame/data.py

```diff
@@ -457,7 +457,13 @@ def groupby_agg(self, by, axis, agg, groupby_args, **kwargs):
 
         # Create new base where all required columns are computed. We don't allow
         # complex expressions to be a group key or an aggeregate operand.
-        assert isinstance(by_frame._op, TransformNode), "unexpected by_frame"
+        allowed_nodes = (FrameNode, TransformNode)
+        if not isinstance(by_frame._op, allowed_nodes):
+            raise NotImplementedError(
```

Comment:
replaced `AssertionError` to `NotImplementedError` so our [QC methods wrapper](https://github.com/modin-project/modin/blob/a43d158ea71e332380f94dad2ed02f1ded870382/modin/experimental/backends/omnisci/query_compiler.py#L95-L102) could catch this and default to pandas automatically